src/interpret/heaptrack_interpret.cpp

```diff
@@ -304,7 +304,36 @@ struct AccumulatedTraceData
         dwfl_end(m_dwfl);
     }
 
-    /// find a file in the sysroot or extra path
+    /// find a file in the sysroot or extra path and add it to the cache
+    const std::string& resolveAndCacheFile(const std::string& fileName)
+    {
+        if (!m_extraPaths.empty()) {
+            // look for the filename, checking both if it's there using the defined path
+            // or just the filename - checking for the full path can be helpful when
+            // there was a local DESTDIR that was used as a staging dir to transfer to
+            // the system that the executable ran on; treating that staging directory as
+            // a second sysroot is easier especially if there are multiple shared objects
+            // with their own debug symbols along with the executable that were all deployed
+            // together
+            const auto baseName = std::filesystem::path(fileName).filename().string();
+            for (const auto& extraPath : m_extraPaths) {
+                auto fileInExtraPath = extraPath + '/' + baseName;
+                if (fileExists(fileInExtraPath))
+                    return m_resolvedFiles.insert(m_resolvedFiles.end(), {fileName, std::move(fileInExtraPath)})
+                        ->second;
+
+                fileInExtraPath = extraPath + '/' + fileName;
+                if (fileExists(fileInExtraPath))
+                    return m_resolvedFiles.insert(m_resolvedFiles.end(), {fileName, std::move(fileInExtraPath)})
```

Comment:
DRY: make this a lambda and use it in both places

Reply:
Resolved this and the other comment about search order together - let me know if that's a step too far; I can back off it it's too complicated of a solution :smile: